pkg/networkservice/common/authorize/server.go

```diff
@@ -28,19 +31,55 @@ import (
 	"github.com/networkservicemesh/sdk/pkg/networkservice/core/next"
 )
 
-type authorizeServer struct{}
+type authorizeServer struct {
+	policy *rego.PreparedEvalQuery
+}
 
 // NewServer - returns a new authorization networkservicemesh.NetworkServiceServers
-func NewServer() networkservice.NetworkServiceServer {
-	return &authorizeServer{}
+func NewServer(p *rego.PreparedEvalQuery) networkservice.NetworkServiceServer {
+	return &authorizeServer{
+		policy: p,
+	}
 }
 
 func (a *authorizeServer) Request(ctx context.Context, request *networkservice.NetworkServiceRequest) (*connection.Connection, error) {
-	// TODO check authorization
+	rs, err := a.policy.Eval(ctx, rego.EvalInput(request.GetConnection().GetPath().GetPathSegments()[request.GetConnection().GetPath().GetIndex()].GetToken()))
+	if err != nil {
+		return nil, status.Error(codes.Internal, err.Error())
+	}
+
+	if err := check(rs); err != nil {
+		return nil, err
+	}
+
 	return next.Server(ctx).Request(ctx, request)
 }
 
 func (a *authorizeServer) Close(ctx context.Context, conn *connection.Connection) (*empty.Empty, error) {
-	// TODO check authorization
+	rs, err := a.policy.Eval(ctx, rego.EvalInput(conn.GetPath().GetPathSegments()[conn.GetPath().GetIndex()].GetToken()))
+	if err != nil {
+		return nil, status.Error(codes.Internal, err.Error())
+	}
+
+	if err := check(rs); err != nil {
+		return nil, err
+	}
+
 	return next.Server(ctx).Close(ctx, conn)
 }
+
+func check(rs rego.ResultSet) error {
+	for _, r := range rs {
+		for _, e := range r.Expressions {
+			t, ok := e.Value.(bool)
+			if !ok {
+				return status.Error(codes.Internal, "non boolean expression")
```

Comment:
can we make this error a little more descriptive? it isn't clear that opa returned a non boolean expression.